Automattic/wp-calypso
# AI Logo generator: clean up and refactor app store

Initially copied from one of Jetpack's blocks, the store is filled with data we don't need and lacking some other that will come in handy.

Related to #85557 

## Proposed Changes

* Use proper naming for store and store properties
* clean up `plans` state prop
* add siteDetails (`data-stores/site`)
* clean up resolver
* add selector for siteDetails, so we don't need to pass down every time (site details is available at quick links as `site`)
* set main component and store tester to use the same props type definition
* update store test component to also display siteDetails
* add loading state to store test as a PoC
* add `suggestions` state prop to hold the suggestions "history" (carousel)

## Testing Instructions

Load the home for one of your sites, look for the quick links and open the "Test Jetpack AI Store" link. You should see 2 `<pre>` boxes, the first with the aiFeatureData retrieved for the site, the second is the `siteDetails` for the site.

Everything else should still work as it was
